ecosystem/helper/src/client/utils/hasGlobalComponent.ts

```diff
@@ -0,0 +1,14 @@
+import { isPlainObject } from '@vuepress/shared'
+import { camelize, capitalize, getCurrentInstance } from 'vue'
+import type { App } from 'vue'
+
+export const hasGlobalComponent = (name: string, app?: App): boolean => {
+  const instance = app ? app._instance : getCurrentInstance()
+
+  return (
+    isPlainObject(instance?.appContext.components) &&
+    (name in instance!.appContext.components ||
+      camelize(name) in instance!.appContext.components ||
+      capitalize(camelize(name)) in instance!.appContext.components)
+  )
+}
```

Comment:
This helper function should be useful enough, as plugins and themes should check if there is already same name global component when registing one to prevent Vue displaying warnings.